testsuite/tests/kuadrant/identical_hostname/test_auth_on_two_routes.py

```diff
@@ -0,0 +1,87 @@
+"""
+Tests behavior of using two HTTPRoutes declaring the same hostname related to AuthPolicy.
+https://github.com/Kuadrant/kuadrant-operator/blob/main/doc/auth.md#limitation-multiple-network-resources-with-identical-hostnames
+(the first topology mentioned there)
+"""
+
+import pytest
+
+from testsuite.policy.authorization.auth_policy import AuthPolicy
+from testsuite.utils import has_condition
+
+pytestmark = [pytest.mark.kuadrant_only]
+
+
+@pytest.fixture(scope="class")
+def authorization2(request, kuadrant, route2, authorization_name2, openshift, label):
+    """2nd Authorization object (In case of Kuadrant AuthPolicy)"""
+    if kuadrant:
+        auth_policy = AuthPolicy.create_instance(openshift, authorization_name2, route2, labels={"testRun": label})
+        auth_policy.authorization.add_opa_policy("rego", "allow = false")
+        request.addfinalizer(auth_policy.delete)
+        auth_policy.commit()
+        auth_policy.wait_for_accepted()
+        return auth_policy
+    return None
+
+
+def test_2nd_authpolicy_rejected(client, authorization, authorization2):
```

Comment:
Since only test names are propagated into ReportPortal, could we make this more descriptive? So the entire test case is deducible from this name alone?

Reply:
`test_identical_hostnames` is probably good enough

Reply:
There will be similar tests for RateLimitPolicy as well so at least `auth` should be part of the name as well. Also, there will be another tests that verify desired behavior (these are for current behavior) so we will end up having quite a few test functions that will need to be nicely named. I think I will go with `test_identical_hostnames_auth_on_routes` and `test_identical_hostnames_auth_on_gw_and_route`

Reply:
Renamed.

Reply:
I think only test method names are propagated, not sure about test file names

Reply:
sorry, my bad. I renamed files instead of test functions :facepalm:

Reply:
Done